cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -208,28 +277,47 @@ public void recordHealthCheck(@NotNull final String executorId, @NotNull final C
                 .setDetails(details)
                 .build()
         );
-        LOGGER.debug("< recordHealthCheck(executorId : {}, details : {})", executorId, protoToString(details));
+        if (LOGGER.isDebugEnabled())
+            LOGGER.debug("< recordHealthCheck(executorId : {}, details : {})", executorId, protoToString(details));
     }
 
     @NotNull
     public List<String> getSeedNodes() {
-        return newArrayList(from(clusterState.executorMetadata()).transform(CassandraFrameworkProtosUtils.executorMetadataToIp()));
+        return newArrayList(from(clusterState.nodes())
+                .filter(new Predicate<CassandraNode>() {
+                    @Override
+                    public boolean apply(CassandraNode v) {
+                        return v.getSeed();
```

Comment:
Give this comparison is a constant, please create either a class with a singleton instance or create a static final field with this implementation.